File: src/main/java/com/ject/studytrip/trip/infra/jpa/DailyGoalRepositoryAdapter.java
package com.ject.studytrip.trip.infra.jpa;

import com.ject.studytrip.trip.domain.model.DailyGoal;
import com.ject.studytrip.trip.domain.repository.DailyGoalRepository;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class DailyGoalRepositoryAdapter implements DailyGoalRepository {
    private final DailyGoalJpaRepository dailyGoalJpaRepository;

    @Override
    public DailyGoal save(DailyGoal dailyGoal) {
        return dailyGoalJpaRepository.save(dailyGoal);
    }

    @Override
    public Optional<DailyGoal> findById(Long id) {
        return dailyGoalJpaRepository.findById(id);
    }
}

File: src/main/kotlin/com/ject/studytrip/trip/infra/jpa/DailyGoalRepositoryAdapter.kt
package com.ject.studytrip.trip.infra.jpa

import com.ject.studytrip.trip.domain.model.DailyGoal
import com.ject.studytrip.trip.domain.repository.DailyGoalRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
class DailyGoalRepositoryAdapter(
    private val dailyGoalJpaRepository: DailyGoalJpaRepository,
) : DailyGoalRepository {
    override fun save(dailyGoal: DailyGoal): DailyGoal = dailyGoalJpaRepository.save(dailyGoal)

    override fun findById(dailyGoalId: Long): Optional<DailyGoal> = dailyGoalJpaRepository.findById(dailyGoalId)
}
